app/services/search_service.rb

```diff
@@ -82,30 +82,28 @@ def search_questions(input, limit = 25, order = :natural, type = :boolean)
   end
 
   # Search nearby nodes with respect to given latitude, longitute and tags
-  def tagNearbyNodes(query, tag, limit = 10)
-    raise("Must separate coordinates with ,") unless query.include? ","
-
-    lat, lon =  query.split(',')
+  def tagNearbyNodes(coordinates, tag, limit = 10)
+    raise("Must contain all four coordinates") if coordinates["nwlat"].nil?
+    raise("Must contain all four coordinates") if coordinates["nwlng"].nil?
+    raise("Must contain all four coordinates") if coordinates["selat"].nil?
+    raise("Must contain all four coordinates") if coordinates["selng"].nil?
 
-    raise("Must have at least one digit after .") unless lat.include? "."
-    raise("Must have at least one digit after .") unless lon.include? "."
+    raise("Must be a float") unless coordinates["nwlat"].is_a? Float
+    raise("Must be a float") unless coordinates["nwlng"].is_a? Float
+    raise("Must be a float") unless coordinates["selat"].is_a? Float
+    raise("Must be a float") unless coordinates["selng"].is_a? Float
 
-    nodes_scope = NodeTag.joins(:tag)
-      .where('name LIKE ?', 'lat:' + lat[0..lat.length - 2] + '%')
+    nodes_scope = NodeTag.joins(:tag).where('name LIKE ?', 'lat%').where('REPLACE(name, "lat:", "") BETWEEN ' + coordinates["selat"].to_s + ' AND ' + coordinates["nwlat"].to_s)
 
     if tag.present?
-      nodes_scope = NodeTag.joins(:tag)
-                           .where('name LIKE ?', tag)
-                           .where(nid: nodes_scope.select(:nid))
+      nodes_scope = NodeTag.joins(:tag).where('name LIKE ?', tag).where(nid: nodes_scope.select(:nid))
```

Comment:
If these could be indented as in the previous version, it makes it a bit more readable. Would you mind, for this and other queries in this PR? Thank you! It helps us keep our code tidy and readable long-term.